File: csharp/bogosort.cs
using System;
static class Program {
	static Random rnd = new Random();
	static void Main(string[] args) {
		BogoSort(GenerateRandomArray(8), 100);
	}
	static void BogoSort(int[] array, int printIterationsEvery) 
	{
		int iteration = 0;
		PrintIteration(array);
		while (!IsSorted(array)) 
		{
			iteration++;
			array = Shuffle(array);
			if (iteration % printIterationsEvery == 0)
				PrintIteration(array, iteration);
		}
		Console.WriteLine("Result:");
		PrintIteration(array, iteration);
	}
	static int[] Shuffle(int[] array) 
	{
		for (int i = 0; i < array.Length; i++) {
			int j = rnd.Next(0, i);
			int temp = array[i];
			array[i] = array[j];
			array[j] = temp;
		}
		return array;
	}
	static bool IsSorted(int[] array) 
	{
		for (int i = 1; i < array.Length; i++)
		{
			if (array[i - 1] > array[i])
				return false;
		}
		return true;
	}
	static int[] GenerateRandomArray(int length)
	{
		int[] array = new int[length];
		for (int i = 0; i < length; i++)
			array[i] = rnd.Next(-100, 100);
		return array;
	}
	static void PrintIteration(int[] array, int iteration = 0) 
	{
		string arrayText = "";
		for (int i = 0; i < array.Length; i++)
		{
			arrayText += array[i];
			if (i != array.Length - 1) arrayText += ", ";
		}
		Console.WriteLine($"{iteration}, [{arrayText}]");
	}
}

File: python/bogosort.py
import random as rnd

def shuffle(list):
    i: int = len(list) - 1
    for num in range(0, i):
        idx: int = rnd.randint(0, i)
        list[num], list[idx] = list[idx], list[num]
    return list

def isSorted(list):
    i: int = len(list)
    for num in range(1, i):
        if list[num - 1] > list[num]:
            return False
    return True

def bogosort(list, printIterationsEvery: int):
    iterations: int = 0
    printIteration(list, iterations)
    while not isSorted(list):
        shuffle(list)
        iterations += 1
        if iterations % printIterationsEvery == 0:
            printIteration(list, iterations)
    print("Result:")
    printIteration(list, iterations)

def printIteration(list, iterations: int):
    print("{}, {}".format(iterations, list))

def generateRandomList(length: int):
    list = []
    for num in range(0, length):
        list.append(rnd.randint(-100, 100))
    return list

bogosort(generateRandomList(8), 100)